File: shell/log.js
#!/usr/bin/env node
var cli = require('cli');
var storage = require('../storage');
var db = storage.getDb();

cli.parse({
    type:   ['t', 'type of action to log'],
    code:   ['c', 'code associated']
});

cli.main(function (args, options) {
    var output = '';
    var accion = '';
    var user = '';
    if (this.argc) {
        // Parse the action
        switch(args[0]) {
            case 'a':
                accion = 'Abrió la puerta';
                break;
            case 'c':
                accion = 'Puerta cerrada';
                break;
            case 'i':
                accion = 'Codigo inválido introducido';
                break;
        }

        // Get the user name                            <td>{{this.codigo}}</td>

        if(args[0] == 'a') {
            var rows = db.run("SELECT user FROM claves WHERE codigo = ?", [args[1]]);
            user = rows[0].user;
        }
        // Async
        db.run("INSERT INTO log (user, accion) VALUES (?, ?)",[user, accion], function(){});
    }
});

File: shell/log.pl
sub run {
	return `./sqlite3 ../data/data.db "@_"`;
}

run("CREATE TABLE IF NOT EXISTS claves(id INTEGER PRIMARY KEY AUTOINCREMENT, codigo INT(4) UNIQUE, user VARCHAR(255), h_desde FLOAT, h_hasta FLOAT);");

run("CREATE TABLE IF NOT EXISTS log(id INTEGER PRIMARY KEY AUTOINCREMENT, user VARCHAR(255), accion VARCHAR(255), date_time DATE_TIME default CURRENT_TIMESTAMP);");

my $accion = '';
my $user = '';

if(@ARGV[0] eq "a"){
    $accion='Abrió la puerta';
	$user = run("select user from claves where codigo = @ARGV[1];");
}
if(@ARGV[0] eq "u"){
    $accion='Desbloqueó la puerta';
	$user = run("select user from claves where codigo = @ARGV[1];");
}
if(@ARGV[0] eq "c"){
    $accion='Puerta cerrada';
}
if(@ARGV[0] eq "i"){
    $accion='Codigo inválido introducido';
}

run("INSERT INTO log (user, accion) VALUES ('$user', '$accion');");
